From d374bdddd4d724f96b93e6a6bb24b55182b32e5c Mon Sep 17 00:00:00 2001
Message-ID: <d374bdddd4d724f96b93e6a6bb24b55182b32e5c.1693948541.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Tue, 5 Sep 2023 22:46:51 +0200
Subject: [PATCH net-next 1/2] net: ethernet: mtk_wed: move mem_region array
 out of mtk_wed_mcu_load_firmware

Remove mtk_wed_wo_memory_region boot struct in mtk_wed_wo. This is a
preliminary patch to introduce WED support for MT7988 SoC.

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 drivers/net/ethernet/mediatek/mtk_wed_mcu.c | 37 ++++++++++-----------
 drivers/net/ethernet/mediatek/mtk_wed_wo.h  |  1 -
 2 files changed, 18 insertions(+), 20 deletions(-)

--- a/drivers/net/ethernet/mediatek/mtk_wed_mcu.c
+++ b/drivers/net/ethernet/mediatek/mtk_wed_mcu.c
@@ -16,12 +16,28 @@
 #include "mtk_wed_wo.h"
 #include "mtk_wed.h"
 
+static struct mtk_wed_wo_memory_region mem_region[] = {
+	[MTK_WED_WO_REGION_EMI] = {
+		.name = "wo-emi",
+	},
+	[MTK_WED_WO_REGION_ILM] = {
+		.name = "wo-ilm",
+	},
+	[MTK_WED_WO_REGION_DATA] = {
+		.name = "wo-data",
+		.shared = true,
+	},
+	[MTK_WED_WO_REGION_BOOT] = {
+		.name = "wo-boot",
+	},
+};
+
 static u32 wo_r32(struct mtk_wed_wo *wo, u32 reg)
 {
 	u32 val;
 
 	if (!wo->boot_regmap)
-		return readl(wo->boot.addr + reg);
+		return readl(mem_region[MTK_WED_WO_REGION_BOOT].addr + reg);
 
 	if (regmap_read(wo->boot_regmap, reg, &val))
 		val = ~0;
@@ -34,7 +50,7 @@ static void wo_w32(struct mtk_wed_wo *wo
 	if (wo->boot_regmap)
 		regmap_write(wo->boot_regmap, reg, val);
 	else
-		writel(val, wo->boot.addr + reg);
+		writel(val, mem_region[MTK_WED_WO_REGION_BOOT].addr + reg);
 }
 
 static struct sk_buff *
@@ -348,18 +364,6 @@ mtk_wed_mcu_load_memory_regions(struct m
 static int
 mtk_wed_mcu_load_firmware(struct mtk_wed_wo *wo)
 {
-	static struct mtk_wed_wo_memory_region mem_region[] = {
-		[MTK_WED_WO_REGION_EMI] = {
-			.name = "wo-emi",
-		},
-		[MTK_WED_WO_REGION_ILM] = {
-			.name = "wo-ilm",
-		},
-		[MTK_WED_WO_REGION_DATA] = {
-			.name = "wo-data",
-			.shared = true,
-		},
-	};
 	const struct mtk_wed_fw_trailer *trailer;
 	const struct firmware *fw;
 	const char *fw_name;
@@ -380,8 +384,8 @@ mtk_wed_mcu_load_firmware(struct mtk_wed
 		 * is defined through reserved memory property.
 		 */
 		wo->boot_regmap = NULL;
-		wo->boot.name = "wo-boot";
-		ret = mtk_wed_get_reserved_memory_region(wo, &wo->boot);
+		ret = mtk_wed_get_reserved_memory_region(wo,
+				&mem_region[MTK_WED_WO_REGION_BOOT]);
 		if (ret)
 			return ret;
 	}
--- a/drivers/net/ethernet/mediatek/mtk_wed_wo.h
+++ b/drivers/net/ethernet/mediatek/mtk_wed_wo.h
@@ -228,7 +228,6 @@ struct mtk_wed_wo_queue {
 
 struct mtk_wed_wo {
 	struct mtk_wed_hw *hw;
-	struct mtk_wed_wo_memory_region boot; /* backward compatibility */
 	struct regmap *boot_regmap;
 
 	struct mtk_wed_wo_queue q_tx;
